nexus/reconfigurator/planning/src/example.rs

```diff
@@ -228,7 +233,15 @@ impl ExampleSystemBuilder {
     /// Currently, this value can be anywhere between 0 and 5. (More can be
     /// added in the future if necessary.)
     pub fn nsleds(mut self, nsleds: usize) -> Self {
-        self.nsleds = nsleds;
+        // Add more sleds if there's a shortfall.
+        if nsleds > self.sled_settings.len() {
+            self.sled_settings.extend(vec![
+                BuilderSledSettings::default();
+                nsleds - self.sled_settings.len()
+            ]);
+        } else if nsleds < self.sled_settings.len() {
+            self.sled_settings.truncate(nsleds);
+        }
```

Comment:
I think all of this could instead be

```rust
self.sled_settings.resize_with(nsleds, BuilderSledSettings::default);
```

or maybe `resize()` instead of `resize_with` if `BuilderSledSettings` is cloneable.